contrib/bundle/test-e2e

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Check for root
+if [[ $EUID -ne 0 ]]; then
+    echo "Please run this script as root"
+    exit 1
+fi
+
+# Assume we're running on this arch
+ARCH=amd64
+
+# Global vars to be used
+# shellcheck source=vars
+source "$(dirname "${BASH_SOURCE[0]}")"/vars
+
+pushd "$ARCHIVE_PATH"
+
+# Untar the bundle
+tar xfvz "$ARCHIVE"
+pushd cri-o
+
+# Install and prepare config
+./install
+
+cat <<EOT >>/etc/crio/crio.conf.d/30-cgroup-manager.conf
+[crio.runtime]
+conmon_cgroup = "pod"
+cgroup_manager = "cgroupfs"
+EOT
+
+# Start CRI-O
+systemctl daemon-reload
+systemctl start crio
+systemctl is-active --quiet crio && echo CRI-O is running
+
+# Start Kubernetes
+GOROOT="$GOROOT_1_16_X64"
+GOPATH="$HOME/go"
+PATH="$GOROOT/bin:$PATH"
+K8SPATH="$GOPATH/src/k8s.io"
+mkdir -p "$K8SPATH"
+pushd "$K8SPATH"
+git clone --depth=1 https://github.com/kubernetes/kubernetes
+pushd kubernetes
+
+echo Downloading server binaries
+export KUBERNETES_SKIP_CONFIRM=1
+export KUBERNETES_SKIP_CREATE_CLUSTER=1
+cluster/get-kube.sh
+
+mkdir -p _output/bin
+tar xfz kubernetes/server/kubernetes-server-linux-amd64.tar.gz
+cp kubernetes/server/bin/{kubectl,kube-apiserver,kube-controller-manager,kube-proxy,kube-scheduler,kubelet} \
+    _output/bin
+
+echo Downloading test binaries
+LATEST=$(curl -sSfL https://dl.k8s.io/ci/latest.txt)
+curl -sSfL "https://dl.k8s.io/ci/$LATEST/kubernetes-test-linux-amd64.tar.gz" \
+    -o- | tar xfz -
+cp kubernetes/test/bin/{ginkgo,e2e.test} _output/bin
+
+export CONTAINER_RUNTIME=remote
+export CGROUP_DRIVER=cgroupfs
+export CONTAINER_RUNTIME_ENDPOINT=/var/run/crio/crio.sock
+export ALLOW_PRIVILEGED=1
+
+IP=$(ip route get 1.2.3.4 | cut -d ' ' -f7 | tr -d '[:space:]')
+echo "Using IP: $IP"
+export DNS_SERVER_IP=$IP
+export API_HOST_IP=$IP
+
+iptables -F
+hack/install-etcd.sh
+export PATH="$GOPATH/src/k8s.io/kubernetes/third_party/etcd:$PATH"
+
+OUTPUT=$(mktemp)
+hack/local-up-cluster.sh -O 2>&1 | tee "$OUTPUT" &
+PID=$!
+
+until grep -q "Local Kubernetes cluster is running" "$OUTPUT"; do
+    echo Waiting for hack/local-up-cluster.sh
+    if ! ps $PID >/dev/null; then
+        exit 1
+    fi
+    sleep 5
+done
+
+echo Cluster is up and running
+
+# Run the tests
+export KUBERUN=/var/run/kubernetes
+export KUBECONFIG=$KUBERUN/admin.kubeconfig
+export PATH="$GOPATH/src/k8s.io/kubernetes/_output/local/bin/linux/amd64:$PATH"
+export KUBE_MASTER_URL=$IP
+export KUBE_MASTER_IP=$IP
+export KUBE_MASTER=$IP
+
+export GINKGO_PARALLEL_NODES=4
+export GINKGO_PARALLEL=y
+
+_output/bin/e2e.test --provider=local --host="https://$IP:6443" \
```

Comment:
node e2e tests can also be run like this, https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/e2e-node-tests.md

Reply:
Although invoking with `e2e.test` might work, I have never tried that way.

Reply:
Assuming crio is up and running, k/k is cloned. I presume the command to invoke the test locally with crio could look something like,

```
export KUBE_SSH_USER=core && make test-e2e-node RUNTIME=remote CONTAINER_RUNTIME_ENDPOINT="unix:///var/run/crio/crio.sock" FOCUS="\[NodeConformance\]|\[NodeFeature:.+\]" SKIP="\[Flaky\]|\[Slow\]|\[Serial\]" TEST_ARGS='--kubelet-flags="--cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/crio.service --non-masquerade-cidr=0.0.0.0/0" --extra-log="{\"name\": \"crio.log\", \"journalctl\": [\"-u\", \"crio\"]}"'
```

Reply:
My goal was to avoid the build time and download the artifacts from the CI bucket. This way we can re-use them directly by `hack/local-up-cluster.sh`.

Reply:
Makes sense.